### PR TITLE
Enable STL assertions for Debug builds with gcc/clang.

### PR DESCRIPTION
**Pull request by endJunction**
This enables assertions in the STL code in the glibcxx.

Also fix a discovered bug in minimum bounding box algorithm.

### CODE REVIEW

File: GeoLib/MinimalBoundingSphere.cpp [hunk 1]
```diff
@@ -165,12 +165,13 @@ MinimalBoundingSphere::recurseCalculation(
         {
             if (i>start_idx)
             {
-                MathLib::Point3d* tmp = sphere_points[start_idx+i];
                 using DiffType = std::vector<MathLib::Point3d*>::iterator::difference_type;
-                std::copy(sphere_points.begin()+static_cast<DiffType>(start_idx),
-                    sphere_points.begin()+static_cast<DiffType>(start_idx + i),
-                    sphere_points.begin()+static_cast<DiffType>(start_idx + 1));
-                sphere_points[start_idx] = tmp;
+                std::vector<MathLib::Point3d*> tmp_ps(
```
Comment (TomFischer) [inline GeoLib/MinimalBoundingSphere.cpp]: I am unsure: Can `tmp_os` be set `const`?

Comment (endJunction) [inline GeoLib/MinimalBoundingSphere.cpp]: @TomFischer Fixed

File: scripts/cmake/CompilerSetup.cmake [hunk 2]
```diff
@@ -48,6 +51,12 @@ if(COMPILER_IS_CLANG)
 		message(FATAL_ERROR "Aborting: Clang 3.3 is required! Found version ${CMAKE_CXX_COMPILER_VERSION}")
 	endif()
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wno-c++98-compat-pedantic")
+	if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+		# Enable assertions in STL in debug mode.
+        message("Set debug flags for clang:")
```
Comment (bilke) [inline scripts/cmake/CompilerSetup.cmake]: Indentation

File: scripts/cmake/CompilerSetup.cmake [hunk 3]
```diff
@@ -48,6 +51,12 @@ if(COMPILER_IS_CLANG)
...
+		set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -D_GLIBCXX_DEBUG -D_GLIBCXX_DEBUG_ASSERT -D_GLIBCXX_DEBUG_PEDASSERT -D_GLIBCXX_DEBUG_VERIFY")
+		message("CMAKE_CXX_FLAGS_DEBUG ${CMAKE_CXX_FLAGS_DEBUG}")
```
Comment (bilke) [inline scripts/cmake/CompilerSetup.cmake]: Is it a debug message? If yes then it can be removed.

Comment (endJunction) [inline scripts/cmake/CompilerSetup.cmake]: Yeah. Forgot to remove it. Thanks!

**Comment by bilke** [comment]
Jenkins, test this please.

**Comment by endJunction** [comment]
Check passed, Mr. Data

**Pull request merged**